uoft-tapp/old-orig-tapp
# Repeatable imports for ChassImporter

Instructions:
- close Docker
- run `docker-compose run rails-app rake db:drop db:create db:migrate db:seed db:seed:chass[round_110]`
  - This will give you the first mock_chass JSON to import

Testing:
- new command `docker-compose run rails-app rake db:seed:chass[file]`, where `file` is any JSON in the `db/seeds` folder
- run the following file to import sequentially to see the changes in the system data
  - `round_110_updated`
  - `round_110_add_apps`
  - `round_112`
- check UI views before running each new file to import to see the changes made

Changes:
- creates new `Application` for an `Applicant` if `Applicant` existed in a previous round
- updates `Applicant`, `Position`, `Application`, and `Instructor` if re-upload from the same round
- delete `Preference` object when applicant un-checks a course after the first upload
- delete `Position` to `Instructor` if instructor is removed from CHASS
- moved `app_id` to `Applicant` model
  - fixed rspec accordingly
  - fix `ChassExporter` accordingly
